src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -10,248 +10,276 @@
+            /// produce a frozen solution for partial semantics, further downstream forking of that solution won't
+            /// rerun generators. This is because of two reasons:
+            /// <list type="number">
+            /// <item>Generally once we've produced a frozen solution with partial semantics, we now want speed rather
+            /// than accuracy; a generator running in a later path will still cause issues there.</item>
+            /// <item>The frozen solution with partial semantics makes no guarantee that other syntax trees exist or
+            /// whether we even have references -- it's pretty likely that running a generator might produce worse results
+            /// than what we originally had.</item>
+            /// </list>
             /// </summary>
-            private abstract class CompilationTrackerState
-            {
-                /// <summary>
-                /// Whether the generated documents in <see cref="GeneratorInfo"/> are frozen and generators should
-                /// never be ran again, ever, even if a document is later changed. This is used to ensure that when we
-                /// produce a frozen solution for partial semantics, further downstream forking of that solution won't
-                /// rerun generators. This is because of two reasons:
-                /// <list type="number">
-                /// <item>Generally once we've produced a frozen solution with partial semantics, we now want speed rather
-                /// than accuracy; a generator running in a later path will still cause issues there.</item>
-                /// <item>The frozen solution with partial semantics makes no guarantee that other syntax trees exist or
-                /// whether we even have references -- it's pretty likely that running a generator might produce worse results
-                /// than what we originally had.</item>
-                /// </list>
-                /// </summary>
-                public readonly bool IsFrozen;
-
-                /// <summary>
-                /// The best compilation that is available that source generators have not ran on. May be an
-                /// in-progress, full declaration, a final compilation.
-                /// </summary>
-                public Compilation CompilationWithoutGeneratedDocuments { get; }
-
-                public CompilationTrackerGeneratorInfo GeneratorInfo { get; }
-
-                protected CompilationTrackerState(
-                    bool isFrozen,
-                    Compilation compilationWithoutGeneratedDocuments,
-                    CompilationTrackerGeneratorInfo generatorInfo)
-                {
-                    IsFrozen = isFrozen;
-                    CompilationWithoutGeneratedDocuments = compilationWithoutGeneratedDocuments;
-                    GeneratorInfo = generatorInfo;
+            public readonly bool IsFrozen;
 
-#if DEBUG
```

Comment:
moved down to subclasses as it can't run during hte constructor now as the base types will not have setup CompilationWithoutGeneratedDocuments  yet.